catalog/transformer/catalog-transformer-nitf/src/test/java/ddf/catalog/transformer/nitf/TestNitfInputTransformer.java

```diff
@@ -0,0 +1,130 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.dynamic.api.MetacardFactory;
+import ddf.catalog.data.dynamic.impl.MetacardFactoryImpl;
+import ddf.catalog.data.dynamic.registry.MetacardTypeReader;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.impl.QueryResponseImpl;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.catalog.transform.CatalogTransformerException;
+
+public class TestNitfInputTransformer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TestNitfInputTransformer.class);
+
+    private static final String BE_NUM_NITF = "/WithBE.ntf";
+
+    private static final String TRE_NITF = "/i_3128b.ntf";
+
+    private static MetacardFactory metacardFactory = null;
+
+    private NitfInputTransformer transformer = null;
+
+    @BeforeClass
+    public static void setupBeforeClass() throws Exception {
+        MetacardFactory mf = new MetacardFactoryImpl();
+        MetacardTypeReader reader = new MetacardTypeReader();
+        reader.setMetacardFactory(mf);
+        InputStream is = Thread.currentThread()
+                .getContextClassLoader()
+                .getResourceAsStream("nitf.xml");
+        reader.parseMetacardDefinition(is);
+        metacardFactory = mf;
+    }
+
+    @Before
+    public void createTransformer()
+            throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+        NitfInputTransformer transformer = new NitfInputTransformer();
+        ddf.catalog.CatalogFramework catalog = mock(ddf.catalog.CatalogFramework.class);
+        when(catalog.query(Matchers.any(QueryRequest.class))).thenReturn(new QueryResponseImpl(null,
+                "sourceId"));
+        transformer.setMetacardFactory(metacardFactory);
+
+        this.transformer = transformer;
+    }
+
+    @Test(expected = CatalogTransformerException.class)
+    public void testNullInput()
+            throws IOException, CatalogTransformerException, UnsupportedQueryException,
+            SourceUnavailableException, FederationException {
+        transformer.transform(null);
+    }
+
+    @Test(expected = CatalogTransformerException.class)
+    public void testBadInput()
+            throws IOException, CatalogTransformerException, UnsupportedQueryException,
+            SourceUnavailableException, FederationException {
+        transformer.transform(new ByteArrayInputStream("{key=".getBytes()));
+    }
+
+    @Test()
+    public void testSorcerWithBE()
+            throws IOException, CatalogTransformerException, UnsupportedQueryException,
+            SourceUnavailableException, FederationException, ParseException {
+        Metacard metacard = transformer.transform(getInputStream(BE_NUM_NITF));
+
+        assertNotNull(metacard);
+
+        assertNotNull(metacard.getCreatedDate());
```

Comment:
This code could be pulled into its own method that takes the metacard and the expected date string as arguments.